safe-global/safe-wallet-web
# fix: Disable submit button if there is a high or critical risk

## What it solves

Resolves #2955 
Resolves #2355 

## How this PR fixes it

- Checks `needsRiskConfirmation` and `isRiskConfirmed` when disabling the sign or execute submit button

## ToDo

- [x] Write tests for `SignForm`
- [x] Write tests for `ExecuteForm`

## How to test it

1. Create a transaction that will result in a high risk e.g. by batching multiple owner changes via the Transaction builder
2. Observe that the Sign/Execute buttons are disabled in that case
3. Confirm accepting the risk via the checkbox
4. Observe the Sign/Execute buttons are enabled

## Screenshots
<img width="1199" alt="Screenshot 2023-12-07 at 13 35 14" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/1545779d-f83e-478a-a8bf-f2f6e607ead2">
<img width="1144" alt="Screenshot 2023-12-07 at 13 35 28" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/a8d865fa-3a6f-4f34-a57b-55fc6ba169c0">
<img width="1128" alt="Screenshot 2023-12-07 at 13 35 38" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/9b1b02cb-157d-4501-8dba-1404f5c6be72">
<img width="1121" alt="Screenshot 2023-12-07 at 13 35 47" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/66aa08b6-b714-42c8-afcf-0a0dc6e25cf1">

## Checklist
